main.tf

```diff
@@ -161,9 +161,9 @@ resource "aws_ram_resource_association" "this" {
 }
 
 resource "aws_ram_principal_association" "this" {
-  count = var.create_tgw && var.share_tgw ? length(var.ram_principals) : 0
+  for_each = var.create_tgw && var.share_tgw ? toset(var.ram_principals) : []
```

Comment:
if any values provided in `var.ram_principals` are computed, this will fail. in addition, the left side of the conditional is of type `toset()` and the right is `tolist()`

Reply:
> if any values provided in `var.ram_principals` are computed, this will fail. 

You have a good point! I will fix it by using a `local` value.

> in addition, the left side of the conditional is of type `toset()` and the right is `tolist()`

Why `tolist()`? The variable `ram_principals` is already a list, moreover `for_each` cannot iterate over lists as you know and that was the reason I used `toset()`... Did I miss something?